RecoJets/JetProducers/plugins/HFJetShowerShape.cc

```diff
@@ -0,0 +1,240 @@
+  template <typename T>
+  void putInEvent(const std::string&, const edm::Handle<edm::View<reco::Jet>>&, std::vector<T>*, edm::Event&) const;
+
+  edm::EDGetTokenT<edm::View<reco::Jet>> jets_token_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_token_;
+
+  //Jet pt/eta thresholds
+  double jetPtThreshold_, jetEtaThreshold_;
+  //HF geometry
+  double hfTowerEtaWidth_, hfTowerPhiWidth_;
+  //Variables for PU subtraction
+  double vertexRecoEffcy_, offsetPerPU_, jetReferenceRadius_;
+  //Pt thresholds for showershape variable calculation
+  double stripPtThreshold_, widthPtThreshold_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HFJetShowerShape::HFJetShowerShape(const edm::ParameterSet& iConfig) {
+  jets_token_ = consumes<edm::View<reco::Jet>>(iConfig.getParameter<edm::InputTag>("theJets"));
+  vertices_token_ = consumes<std::vector<reco::Vertex>>(iConfig.getParameter<edm::InputTag>("theVertices"));
+  jetPtThreshold_ = iConfig.getParameter<double>("jetPtThreshold");
+  jetEtaThreshold_ = iConfig.getParameter<double>("jetEtaThreshold");
+  hfTowerEtaWidth_ = iConfig.getParameter<double>("hfTowerEtaWidth");
+  hfTowerPhiWidth_ = iConfig.getParameter<double>("hfTowerPhiWidth");
+  vertexRecoEffcy_ = iConfig.getParameter<double>("vertexRecoEffcy");
+  offsetPerPU_ = iConfig.getParameter<double>("offsetPerPU");
+  jetReferenceRadius_ = iConfig.getParameter<double>("jetReferenceRadius");
+  stripPtThreshold_ = iConfig.getParameter<double>("stripPtThreshold");
+  widthPtThreshold_ = iConfig.getParameter<double>("widthPtThreshold");
+
+  produces<edm::ValueMap<float>>("sigmaEtaEta");
+  produces<edm::ValueMap<float>>("sigmaPhiPhi");
+  produces<edm::ValueMap<int>>("centralEtaStripSize");
+  produces<edm::ValueMap<int>>("adjacentEtaStripsSize");
+}
+
+HFJetShowerShape::~HFJetShowerShape() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void HFJetShowerShape::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  //Jets
+  edm::Handle<edm::View<reco::Jet>> theJets;
+  iEvent.getByToken(jets_token_, theJets);
+
+  //Vertices
+  edm::Handle<std::vector<reco::Vertex>> theVertices;
+  iEvent.getByToken(vertices_token_, theVertices);
+  int nPV = theVertices->size();
+
+  //Products
+  std::vector<float>* v_sigmaEtaEta = new std::vector<float>;
+  std::vector<float>* v_sigmaPhiPhi = new std::vector<float>;
+  std::vector<int>* v_size_CentralEtaStrip = new std::vector<int>;
+  std::vector<int>* v_size_AdjacentEtaStrips = new std::vector<int>;
+
+  //Et offset for HF PF candidates
+  double PUoffset = offsetPerPU_ / (TMath::Pi() * jetReferenceRadius_ * jetReferenceRadius_) * nPV / vertexRecoEffcy_ *
```

Comment:
```suggestion
  double puOffset = offsetPerPU_ / (M_PI * jetReferenceRadius_ * jetReferenceRadius_) * nPV / vertexRecoEffcy_ *
```
- pi is known from elsewhere (and remove TMath include)
- variable names should not be capitalized